storage/remote/queue_manager.go

```diff
@@ -1186,18 +1202,18 @@ func (s *shards) runShard(ctx context.Context, shardID int, queue *queue) {
 			timer.Reset(time.Duration(s.qm.cfg.BatchSendDeadline))
 
 		case <-timer.C:
-			// We need to take the lock when getting a batch to avoid
+			// We need to take the write lock when getting a batch to avoid
```

Comment:
A comment as to why this needs to be the shard write mutex instead of the queue batch mutex would be useful.

Reply:
I expanded some on this comment for the reasoning here. Let me know if it addresses your concerns!